src/cli/entry_type.rs

```diff
@@ -0,0 +1,114 @@
+use std::str::FromStr;
+
+use colored::Colorize;
+use convert_case::Case;
+use structopt::StructOpt;
+
+use crate::{
+    file_tree::{build_file_tree, load_directory_into_memory, FileTree},
+    scaffold::{
+        dna::DnaFileTree,
+        entry_type::{crud::Crud, definitions::FieldDefinition, scaffold_entry_type},
+        zome::ZomeFileTree,
+    },
+    templates::ScaffoldedTemplate,
+    utils::{check_case, input_with_case},
+};
+
+#[derive(Debug, StructOpt)]
+/// Scaffold an entry type and CRUD functions into an existing zome
+pub struct EntryType {
+    #[structopt(long)]
+    /// Name of the dna in which you want to scaffold the zome
+    pub dna: Option<String>,
+
+    #[structopt(long)]
+    /// Name of the integrity zome in which you want to scaffold the entry definition
+    pub zome: Option<String>,
+
+    /// Name of the entry type being scaffolded
+    pub name: Option<String>,
+
+    #[structopt(long)]
+    /// Whether this entry type should be referenced with its "EntryHash" or its "ActionHash"
+    /// If referred to by "EntryHash", the entries can't be updated or deleted
+    pub reference_entry_hash: Option<bool>,
+
+    #[structopt(long, parse(try_from_str = Crud::from_str))]
+    /// The Create, "Read", "Update", and "Delete" zome call functions that should be scaffolded for this entry type
+    /// If "--reference-entry-hash" is "true", only "Create" and "Read" will be scaffolded
+    pub crud: Option<Crud>,
+
+    #[structopt(long)]
+    /// Whether to create a link from the original entry to each update action
+    /// Only applies if update is selected in the "crud" argument
+    pub link_from_original_to_each_update: Option<bool>,
+
+    #[structopt(long, value_delimiter = ",", parse(try_from_str = FieldDefinition::from_str))]
+    /// The fields that the entry type struct should contain
+    /// Syntax: <FIELD_NAME>:<FIELD_TYPE>:<WIDGET>:<LINKED_FROM> , (widget and linked_from are optional)
+    /// Eg. "title:String:TextField" , "posts_hashes:Vec\<ActionHash\>::Post"
+    pub fields: Option<Vec<FieldDefinition>>,
+
+    #[structopt(long)]
+    /// Skips UI generation for this entry-type, overriding any specified widgets in the --fields option.
+    pub no_ui: bool,
+}
+
+impl EntryType {
+    pub fn run(self, template_file_tree: FileTree) -> anyhow::Result<()> {
+        let current_dir = std::env::current_dir()?;
+        let file_tree = load_directory_into_memory(&current_dir)?;
+        let name = match self.name {
+            Some(n) => {
+                check_case(&n, "entry type name", Case::Snake)?;
+                n
+            }
+            None => input_with_case("Entry type name (snake_case):", Case::Snake)?,
+        };
+
+        let dna_file_tree = DnaFileTree::get_or_choose(file_tree, self.dna.as_deref())?;
+        let zome_file_tree =
+            ZomeFileTree::get_or_choose_integrity(dna_file_tree, self.zome.as_deref())?;
+
+        if self.no_ui {
+            let warning_text = r#"
```

Comment:
I did not see this text being printed out at all when I tested it with the `--no-ui` flag. Also, it looks like the interactive choice of whether or not to scaffold the entry type with UI comes only later, as part of the `scaffold_entry_type()` function if I'm not mistaken. So maybe the warning should get moved into that function instead. And I think it would be somewhat important that there is still the possibility to cancel the scaffolding of the entry-type at the moment this warning appears.

Reply:
mmmh, wasn't able to reproduce it, the text get's printed on my end.

But I see the value of having the warning moved to the `scaffold_entry_type` function

Reply:
Oh I see :+1:

Reply:
Actually, in light of this misunderstanding I'm starting to think that the warning should be moved outside of the function again and be displayed even before asking the entry name on [this line](https://github.com/c12i/scaffolding/blob/9ad5fd5e025708082cd8b4b54c435eda9dfc0ce9/src/cli/entry_type.rs#L62).
And on my Ubuntu terminal the warning is more brownish than yellow and therefore seems more like a comment, maybe `bright_wellow()` would provide a bit more contrast? It's possible that the warning was actually printed and I just didn't notice it because it did not appear at the moment when I expected it (I expected it to appear in the interactive part where it asks whether the entry field should be shown in the UI).
![image](https://github.com/holochain/scaffolding/assets/36768177/ac5fe857-1acb-4e3e-8dae-3c53aecb7311)